pkg/operator/staticpod/controller/staticpodstate/staticpodstate_controller.go

```diff
@@ -127,10 +132,8 @@ func (c *StaticPodStateController) sync(ctx context.Context, syncCtx factory.Syn
 
 	switch {
 	case len(images) == 0:
-		// if NodeStatuses is still empty, we are most probably in bootstrapping phase and this controller races with the
-		// installer controller. Hence, ignore that case. It will settle.
-
-		if len(originalOperatorStatus.NodeStatuses) > 0 {
+		// only generate an event if we found pods, but none has an image.
+		if podsFound {
```

Comment:
is this variable necessery? at this point `images` cannot be `0` (line 95)

Reply:
that was the bug, if the loop can't iterate over the nodestatus or the pods (because they don't exist or whatever), images will be 0
Another thing Stefan and I discussed is that doesn't seem to be possible having a pod without an image set :shrug: 
https://github.com/openshift/library-go/pull/1203#discussion_r705286326

Reply:
ic, so we are playing safe here.

Reply:
right